### PR TITLE
Add packaging support for unix domain sockets

### PR DESCRIPTION
**Pull request by minfrin**
- Add /run/jenkins path in systemd-tmpfiles as a home for the socket
- Support JENKINS_UNIX_DOMAIN_PATH variable to pass the socket path

Added to winstone: https://github.com/jenkinsci/winstone/pull/346 / https://github.com/jenkinsci/winstone/issues/345

### Testing done

- jenkins.war from most recent master

- RPM created as follows:

```
WAR=/home/minfrin/jenkins.war MSI= BUILDENV=env/release.mk BRAND=branding/jenkins.mk make rpm
```

- /etc/sysconfig/jenkins modified for testing as follows:

```
JAVA_HOME=/etc/alternatives/jre_17

JENKINS_PORT=-1
JENKINS_PREFIX=/jenkins
JENKINS_UNIX_DOMAIN_PATH=/var/run/jenkins/jenkins.socket
```

Apache httpd configured as follows:

```
  ProxyPass unix:///run/jenkins/jenkins.socket|http://localhost/jenkins nocanon
  ProxyPassReverse unix:///run/jenkins/jenkins.socket|http://localhost/jenkins
```

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by timja — APPROVED** [review]
Thanks!

**Comment by timja** [comment]
cc @kmartens27 / @MarkEWaite for changelog

File: systemd/jenkins.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+D /run/jenkins 0770 jenkins jenkins -
```
Comment (Artturin) [inline systemd/jenkins.conf:1]: `RuntimeDirectory="jenkins"`

in `jenkins.service` should be a better fit here than a tmpfile.

Comment (MarkEWaite) [reply systemd/jenkins.conf:1]: > in `jenkins.service` should be a better fit here than a tmpfile.

It is usually much more effective to submit a pull request  with an explanation why a different location would be better.  If you're not ready to submit a pull request, then an issue report is another way to start the conversation.

Commenting on a merged pull request is not nearly as likely to result in any action.

**Pull request merged**